api/nakadi-event-bus-api.yaml

```diff
@@ -1503,11 +1541,37 @@ definitions:
         type: string
         default: 'random'
 
+      compatibility_mode:
+        description: |
+          Compatibility modes provide a mean for event owners to evolve their schema, given changes
+          respect the semantics defined by this field.
+
+          It's designed to be flexible enough so that producers can evolve their schemas while not
+          inadvertently breaking existent consumers.
+
+          Once defined, the compatibility mode is fixed, since otherwise it would break a predefined contract,
+          declared by the producer.
+
+          List of available compatibility modes:
+
+          - 'deprecated': Schema changes are not allowed. This is used only by legacy event types. This option is not
+            available for new event types. Schema validation is lenient and allow fields that are not defined. Also,
+            it's possible to use the full json schema specification for defining schemas.
+
+          - 'none': Any schema modification is accepted, even if it might break existing producers or consumers.
+
+          - 'compatible': Schema changes never break existing consumers. Every event published since the first version
+            is still valid based on the newest schema. When in compatible mode, it's allowed to add new optional
+            properties and definitions to an existing schema, but no other changes are allowed. If additionalProperties
```

Comment:
Couldn't we prohibit additionalProperties altogether here? In the end, if events are to be archived in the Data Lake additionalProperties can become a huge pain. (E.g. types can change freely any time. And it is hard to imagine how this would integrate with a system that stores security labels or documentation for event attributes.) Is there really a good use case for this?